### PR TITLE
Port datasets controller to datablock storage

### LINKED ISSUE #56998: Port datasets controller to Datablock Storage
We might not switch to the ported version yet since we don't want to have a dual authority confusion. The plan is to keep firebase as authoritative until we get to post-firebase-cleanup #56994 .

### PR DESCRIPTION
**Pull request by snickell**
The datasets controller allows editing the standard shared tables available students. Its used by content editors mainly. This ports the datasets controller to editing our datablock storage controller objects.

Methods/behaviors that we wanted to share with datablock_storage_controller.rb were migrated to datablock_storage_table.rb object so they're easy to share.

Also adds a link from the datasets controller to the manifest controller

Fixes #56998

Co-authored-by: Cassi Brenci <cnbrenci@users.noreply.github.com>

### CODE REVIEW

**Comment by snickell** [comment]
I ported levels_controller.rb to datablock storage as well, because it has a backlink to https://github.com/code-dot-org/code-dot-org/issues/56998, but I couldn't figure out how to test it, these lines:
https://github.com/code-dot-org/code-dot-org/blob/e5238bab4207ecc61f7d12d1f49f87eda402ead4/dashboard/app/controllers/levels_controller.rb/#L161-L163
```
    if @level.is_a? Applab
      # TODO: unfirebase, this is migrated to datablock storage but NOT TESTED: #56998
      # TODO: post-firebase-cleanup, remove this code: #56994
      @dataset_library_manifest = DatablockStorageLibraryManifest.instance.library_manifest
    end
```

Other than that, I think this might be ready to ship!!

**Comment by snickell** [comment]
It /might/ be nice to add a courtesy link from /datasets (e.g. http://localhost-studio.code.org:3000/datasets) to /datasets/manifest/edit (http://localhost-studio.code.org:3000/datasets/manifest/edit)

**Comment by snickell** [comment]
One failure still:
```
FAIL["test_update_manifest:_passes_through_the_error_code", "DatasetsControllerTest", 1405.6563193559998]
--
3950 | test_update_manifest:_passes_through_the_error_code#DatasetsControllerTest (1405.66s)
3951 | Expected response to be a <401: unauthorized>, but was a <200: OK>
3952 | Response body: {}.
3953 | Expected: 401
3954 | Actual: 200
3955 | test/controllers/datasets_controller_test.rb:48:in `block in <class:DatasetsControllerTest>'
3956 | test/testing/setup_all_and_teardown_all.rb:36:in `run'
```

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -158,10 +158,9 @@ def edit
     @in_script = @level.script_levels.any? || any_parent_in_script
     @standalone = ProjectsController::STANDALONE_PROJECTS.values.map {|h| h[:name]}.include?(@level.name)
     if @level.is_a? Applab
-      # TODO: unfirebase, migrate this to datablock storage, ok to be migrated but not enabled: #56998
+      # TODO: unfirebase, this is migrated to datablock storage but NOT TESTED: #56998
       # TODO: post-firebase-cleanup, remove this code: #56994
-      fb = FirebaseHelper.new('shared')
-      @dataset_library_manifest = fb.get_library_manifest
+      @dataset_library_manifest = DatablockStorageLibraryManifest.instance.library_manifest
     end
```
Comment (snickell) [inline dashboard/app/controllers/levels_controller.rb:162]: How do we test this is working?

Comment (cnbrenci) [reply dashboard/app/controllers/levels_controller.rb:162]: Tested in level editor that after updating the manifest, we saw the updated manifest in level editor!

**Review by cnbrenci — APPROVED** [review]
Great work! 🚀 

FYI for @kakiha11 that once this goes to prod, we'll be editing the dataset manifest on the datablock storage side. Any changes to the manifest will have to be synced back to firebase if we want them available for everyone right away.

**Pull request merged**